hazelcast/src/main/java/com/hazelcast/cache/impl/record/CacheRecordHashMap.java

```diff
@@ -36,24 +34,36 @@
     private static final long serialVersionUID = 1L;
 
     private final transient CacheContext cacheContext;
+    private boolean entryCountingEnable;
 
     public CacheRecordHashMap(int initialCapacity, CacheContext cacheContext) {
         super(initialCapacity);
         this.cacheContext = cacheContext;
     }
 
-    public CacheRecordHashMap(int initialCapacity, float loadFactor, int concurrencyLevel,
-            ConcurrentReferenceHashMap.ReferenceType keyType,
-            ConcurrentReferenceHashMap.ReferenceType valueType,
-            EnumSet<Option> options, CacheContext cacheContext) {
-        super(initialCapacity, loadFactor, concurrencyLevel, keyType, valueType, options);
-        this.cacheContext = cacheContext;
+    // Called by only same partition thread. So there is no synchronization and visibility problem.
+    @Override
+    public void setEntryCounting(boolean enable) {
+        if (enable) {
+            if (!entryCountingEnable) {
+                // It was disable before but now it will be enable.
+                // Therefore, we increase the entry count as size of records.
+                cacheContext.increaseEntryCount(size());
+            }
+        } else {
+            if (entryCountingEnable) {
+                // It was enable before but now it will be disable.
+                // Therefore, we decrease the entry count as size of records.
+                cacheContext.decreaseEntryCount(size());
```

Comment:
The decrease part is called on source node after migration is committed. (on rollback migration since there is no primary state change, this code block should not be executed because it expects `enable != entryCountingEnable`) 

I mean the node was the owner of the partition but now it is not owner anymore. But in fact this part is called after `clearPartitionReplica` so when this decrease part is called, there is not record in cache record map because it is already cleared and its counts are already decreased from entry count. 

So in fact this part should not have any effect but I had added it to be consistent with increase part.